src/core/SampleDecoder.cpp

```diff
@@ -25,6 +25,38 @@ std::vector<SampleDecoder::Decoder> SampleDecoder::s_decoders = {
 	&SampleDecoder::decodeSampleDS
 };
 
+auto SampleDecoder::supportedAudioTypes() -> std::vector<AudioType>
+{
+	auto sfFormatInfo = SF_FORMAT_INFO{};
+
+	// Add DrumSynth by default since that support comes from us
+	auto audioTypes = std::vector<AudioType>{AudioType{"DrumSynth", "ds"}};
+
+	auto simpleTypeCount = 0;
+	sf_command(nullptr, SFC_GET_SIMPLE_FORMAT_COUNT, &simpleTypeCount, sizeof(int));
+
+	// TODO: Ideally, this code should be iterating over the major formats, but some important extensions such as *.ogg
+	// are not included. This is planned for future versions of sndfile.
+	for (int simple = 0; simple < simpleTypeCount; ++simple)
+	{
+		sfFormatInfo.format = simple;
+		sf_command(nullptr, SFC_GET_SIMPLE_FORMAT, &sfFormatInfo, sizeof(sfFormatInfo));
+
+		auto it = std::find_if(audioTypes.begin(), audioTypes.end(),
+			[&](const AudioType& type) { return sfFormatInfo.extension == type.extension; });
+		if (it != audioTypes.end()) { continue; }
+
+		auto name = std::string{sfFormatInfo.extension};
```

Comment:
I think the `const char*` strings the sndfile API returns in the `SF_FORMAT_INFO` object have a static lifetime, so you probably could store them in a `std::string_view` if you wanted. And I assume the values are never null either?

Reply:
I used `std::string` to first transform the extension name to be uppercase, and I figured that would make it lose its static duration. I did not want to use the name provided by sndfile because it's oddly specific for various formats, when all I need is the extension as the name (uppercased to make it look nicer for the file dialog).

And this is a bit of a side bar, but sndfile is confusing with this. They have major formats and sub formats, as well as simple formats. Subformats can return null for the extension, while that doesn't happen for major and simple formats. I think ideally we should be using the major formats, but the extensions for them are not.. reliable. For example, the extension for the major format of OGG is "oga", but should also consist of "ogg". Simple formats however cover the most common file types, so I went with that. Sndfile plans to support multiple file extensions for major formats in the future though.